### PR TITLE
Cortx-33443 Test Automation TEST-47158, TEST-47161, TEST-47162

### PR DESCRIPTION
**Pull request by priyankaborawake**
# Problem Statement
- Cortx-33443 Test Automation TEST-47158, TEST-47161, TEST-47162

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by priyankaborawake** [comment]
[collections.log](https://github.com/Seagate/cortx-test/files/9555550/collections.log)

File: libs/dtm/dtm_recovery.py [hunk 1]
```diff
@@ -345,6 +345,35 @@ def multi_process_restart_with_delay(self, master_node, health_obj, pod_prefix,
                 self.log.info("Process %s restarted successfully", process)
         return True
 
+    def process_restart_mp(self, master_node, pod_selected, container,
```
Comment (sonalk0209) [inline libs/dtm/dtm_recovery.py:348]: don't we already process_restart_with_delay so what is different for this one?

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 2]
```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
...
         self.test_completed = True
         self.log.info("ENDED: Verify delete after multiple m0d restart using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm
+    @pytest.mark.tags("TEST-47158")
+    def test_continuous_write_read_during_k_m0d_restart_overlapping(self):
+        """
+        Verify Continuous Write/Reads during K process restart.
+        (Different m0d processes to be selected one after another with delay of 2 mins till k,
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:313]: so this is again for K and not based on formula that we discussed recentely?

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 3]
```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
...
+        proc_read_op.start()
+        self.log.info("Step 3b: Start write Operations in loop in background:")
+        args = {'bucket_prefix': self.bucket_name, 'object_prefix': self.object_name,
+                'no_of_clients': self.test_cfg['clients'],
+                'no_of_samples': self.test_cfg['samples'], 'log_file_prefix': log_file_prefix,
+                'queue': que_2, 'loop': self.test_cfg['loop_count'], 'created_bucket': bucket_list}
+        proc_write_op = multiprocessing.Process(target=self.dtm_obj.perform_write_op, kwargs=args)
+        proc_write_op.start()
+
+        time.sleep(self.delay)
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:356]: add a logger whenever we are sleeping for some time so test doesn't look like stuck

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 4]
```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
...
+
+        self.log.info("Step 4: Perform m0d Process Restart on %s unique data pods during"
+                      " Write and Read Operations",self.kvalue)
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:359]: space after ,

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 5]
```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
...
+                continue
+            else:
+                pods_list.append(pod_selected)
+                count = count+1
+            proc = multiprocessing.Process(target = self.dtm_obj.process_restart_mp,
+                                           args = (self.master_node_list[0],pod_selected,container,
+                                                   self.m0d_process, 300, m0d_que))
+            proc.start()
+            m0d_proc_list.append(m0d_proc_list)
+            time.sleep(120)
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:377]: same for all sleeps, plz log it

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 6]
```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
...
+        self.log.info("Sleep for 300+120 seconds")
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:378]: if we have config values for these sleep times, better use those

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 7]
```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
...
+
+        self.log.info("Step 5: Verify hctl status if all services are online")
+        resp = self.ha_obj.poll_cluster_status(self.master_node_list[0], 300)
+        assert_utils.assert_true(resp[0],resp[1])
+
+        self.log.info("Step 6a: Wait for READ Operation to complete.")
+        if proc_read_op.is_alive():
+            proc_read_op.join()
+        resp = que_1.get()
+        assert_utils.assert_true(resp[0], resp[1])
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:395]: I don't see retry count getting mentioned, so aren't we expecting failures at least when m0d gets down?

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 8]
```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
...
+            else:
+                pods_list.append(pod_selected)
+                count = count + 1
+            proc = multiprocessing.Process(target=self.dtm_obj.process_restart_mp,
+                                           args=(self.master_node_list[0], pod_selected, container,
+                                                 self.m0d_process, 1200, m0d_que))
+            proc.start()
+            m0d_proc_list.append(m0d_proc_list)
+
+        self.log.info("Sleep for 1200+120 seconds")
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:461]: similar comments as above

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 9]
```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
...
+        self.log.info("ENDED:  Verify continuous WRITE and READ during different m0d restart "
+                      " using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm
+    @pytest.mark.tags("TEST-47163")
+    def test_continuous_write_read_during_k_m0d_restart(self):
+        """
+        Verify Continuous Write/Reads during K process restart.
+        (Randomly select m0d process, kill it ,sleep for 5 mins, start it and.Repeat this K times)
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:572]: so we have same process getting killed here if diff then the only diff between 1st and this is, in 1st we are killing processes after every 2 mins and restart after 5 and here its one by one. Is that right?

**Comment by priyankaborawake** [comment]
[collection.log](https://github.com/Seagate/cortx-test/files/9630773/collection.log)

**Review by sonalk0209 — APPROVED** [review]
Approving for merge!!!

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**